checkov/secrets/scan_git_history.py

```diff
@@ -92,27 +95,32 @@ def _create_secret_collection(self) -> None:
                 self.secrets[key].add(secret_data["potential_secret"])
         logging.info(f"Created secret collection for {len(self.history_store.secrets_by_file_value_type)} secrets")
 
-    @time_it
+    def set_repo(self, root_folder: str | None = None) -> bool:
+        if not root_folder:
+            root_folder = self.root_folder
+        if git_import_error is not None:
+            logging.warning(f"Unable to load git module (is the git executable available?) {git_import_error}")
+            return False
+        try:
+            self.repo = Repo(root_folder)
+            return True
+        except Exception as e:
+            logging.error(f"Folder {root_folder} is not a GIT project {e}")
+            return False
+
     def _get_commits_diff(self, root_folder: str, last_commit_sha: Optional[str] = None) -> List[Commit]:
```

Comment:
sounds like we no longer need root folder in this function since it has it in self